app/models/agent.rb

```diff
@@ -61,8 +61,8 @@ class Agent < ActiveRecord::Base
   has_many :scenario_memberships, :dependent => :destroy, :inverse_of => :agent
   has_many :scenarios, :through => :scenario_memberships, :inverse_of => :agents
 
-  scope :active,   -> { where(disabled: false) }
-  scope :inactive, -> { where(disabled: true) }
+  scope :active,   -> { joins(:user).where('users.deactivated_at IS NULL').where(disabled: false) }
+  scope :inactive, -> { joins(:user).where(["users.deactivated_at IS NOT NULL OR disabled = ?", true]) }
```

Comment:
Hopefully this will be fast enough.  I've had trouble with disjoint SQL statements in the past, in which case we may need a cache column.